naomi-setup.sh fixes to Naomi-Dev

aaronchantrill opened the pull request:
## Description
This is a duplicate of PR #137 which fixes a few dependency issues and fixes a couple of features that got left out of the first version of Naomi 2.2

## Related Issue
[#136 Naomi-setup.sh issues](https://github.com/NaomiProject/Naomi/issues/136)

## Motivation and Context
This is a hotfix to fix a missing requirement when building Naomi using the naomi-setup.py script.

## How Has This Been Tested?
It has been tested on Raspbian 2018-11-13 running on a Raspberry Pi 3B+.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

AustinCasteel committed:
Update issue templates

aaronchantrill committed:
Fixed a few issues with Naomi 2.2

Added the generated Naomi file that uses the local python to the
.gitignore list

Modified naomi-setup.sh and added libncurses5-dev to the list
of apt packages to install

Removed nano and git from python-requirements.txt (duplicating
PR # 131 by redragonx)

Added a check for audio engines to populate.py that runs directly
after the language selection, so the user doesn't get to the end
and suddenly discover they can't complete the install.

Fixed an issue with sphinxvocab.py where temporary files were
not being removed. This was for debugging purposes.

AustinCasteel reviewed (commented):
The changes to the files on their own look fine. However in running tests I got two things that came up. [Pop error could just be because I did not do the sphinx config before running setup. However the setup traceback could be an issue.](https://gist.github.com/Longshotpro2/5b81abdc1afd8fdd9eaca0c462204c4e)

aaronchantrill committed:
More fixes for Naomi setup

Changed downloaded setuptools version to latest

Changed plugin search path for audiotools so it
no longer generates warnings from other plugins

AustinCasteel reviewed (approved):
Traceback fixed and verified working on RPI!
